lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -404,22 +404,22 @@ impl ModuleCodeGenerator<X64FunctionCode, X64ExecutionContext, CodegenError>
         mut self,
         _: &ModuleInfo,
     ) -> Result<(X64ExecutionContext, Box<dyn CacheGen>), CodegenError> {
-        let (assembler, breakpoints) = match self.functions.last_mut() {
-            Some(x) => (x.assembler.take().unwrap(), x.breakpoints.take().unwrap()),
-            None => {
-                return Err(CodegenError {
-                    message: "no function",
-                });
-            }
```

Comment:
@losfair the addition below is repeated in `next_function`, maybe it could be consolidated in the future.